Sources/PushNotifications.swift

```diff
@@ -190,6 +182,28 @@ import Foundation
         return persistenceService.getSubscriptions()
     }
 
+    /**
+     Handle Remote Notification.
+
+     - Parameter userInfo: Remote Notification payload.
+     */
+    @objc public func handleNotification(userInfo: [AnyHashable: Any]) {
+        #if os(iOS)
+            let applicationState = UIApplication.shared.applicationState
+            let eventType = (applicationState == .inactive) ? ReportEventType.Open.rawValue : ReportEventType.Delivery.rawValue
+        #elseif os(OSX) //TODO: Needs more investigation.
```

Comment:
There's no `applicationState` in `NSApplication`. This needs to be investigated how we will track open push notifications for macOS.